packages/components/src/components/horizontal-nav/horizontal-nav.component.ts

```diff
@@ -0,0 +1,196 @@
+   */
+  @state() protected priorityMenuItems: SynNavItem[] = [];
+
+  /**
+   * Get a list of all slotted `<syn-nav-item />` elements
+   */
+  private getSlottedNavItems() {
+    const slottedItems = Array.from(
+      this.defaultSlot.assignedElements({ flatten: true }),
+    ) as HTMLElement[];
+
+    return slottedItems.filter(i => i instanceof SynNavItem) as SynNavItem[];
+  }
+
+  /**
+   * Adjust the visual placement of the priority menu
+   * @param startPos The start position to calculate from
+   */
+  private adjustPriorityMenuPosition(startPos: number) {
+    const relativeX = startPos - this.priorityMenu.offsetLeft;
+    this.priorityMenu.style.transform = `translate(${relativeX}px, 0)`;
+  }
+
+  /**
+   * Determines which items should be shown or hidden, depending on the current width
+   */
+  private handlePriorityMenu() {
+    const navItems = this.getSlottedNavItems();
+    const { width } = this.navItemWrapper.getBoundingClientRect();
+
+    let firstHiddenItemRightPos: number | undefined;
+    const nextPriorityMenuItems: SynNavItem[] = [];
+
+    // eslint-disable-next-line complexity
+    navItems.forEach(item => {
+      // If we already have found an item,
+      // skip the position check as we already know the first hidden item.
+      // Else, check if the most right position of the item is in the view
+      const isHidden = firstHiddenItemRightPos || item.getBoundingClientRect().right > width;
+
+      // eslint-disable-next-line no-param-reassign
+      item.style.visibility = isHidden ? 'hidden' : 'visible';
+      item.setAttribute('aria-hidden', isHidden ? 'true' : 'false');
+
+      if (isHidden) {
+        // Save a reference to the currently hidden elements.
+        // This makes it possible to render them via renderPriorityMenu
+        nextPriorityMenuItems.push(item);
+
+        // Get the position of the first item
+        // Will get used to position the priority menu
+        if (!firstHiddenItemRightPos) {
+          firstHiddenItemRightPos = item.offsetLeft;
+        }
+      }
+    });
+
+    this.priorityMenuItems = nextPriorityMenuItems;
+
+    // Only show the priority menu label if there are no visible items anymore
+    const labelVisibility = nextPriorityMenuItems.length === navItems.length ? 'visible' : 'hidden';
+    this.priorityMenuLabelElement.style.visibility = labelVisibility;
+
+    if (firstHiddenItemRightPos) {
+      this.adjustPriorityMenuPosition(firstHiddenItemRightPos);
+    }
+  }
+
+  /**
+   * Renders the priority menu with all items that are currently not available on screen
+   */
+  private renderPriorityMenu() {
+    const items = this.priorityMenuItems;
+
+    return html`
+      <div
+        class=${classMap({
+          'priority-menu': true,
+          'priority-menu--hidden': items.length === 0,
+        })}
+        part="priority-menu"
+      >
+        <syn-nav-item>
+          <syn-icon name="more_horiz" label="More" slot="prefix"></syn-icon>
```

Comment:
`more_horiz`? :D Is this the original name?  OMG. This is really the original material name...

This should be a system icon, right? And maybe we can name it nicer then...

Reply:
Yeah, it unfortunately is the "currect" name :). I am unsure if we may get problems with the renaming. Currently, it is pretty consistent. However, I personally have no problems giving it another name. We could do the same as with the `caret`, which we use to transform between left/right/top/bottom and also exists in the material library. So giving it a name of `more` and just toggling it between horizontal and vertical mode would be fine for me (at least if the original icons are the same with different orientation)